src/store/store.js

```diff
@@ -2,82 +2,112 @@ import {makeAutoObservable} from 'mobx';
 
 class Store {
   BASE_URL = 'http://192.168.100.106:1339';
+
   categories = [];
+
   currentCategory = [];
+
   questionSets;
+
   chosenQuestionsSet = [];
+
   questions;
+
   currentQuestion = [];
+
   currentAnswers = [];
+
   chosenMode = '';
+
   showHelpDialog = false;
+
   isChooseModeDialog = false;
 
+  isAnswerHasChosen = false;
+
   constructor() {
-    (async () => {
-      await this.fetchCategories();
-      await this.fetchQuestionSets();
-      await this.fetchQuestions();
-    })();
+    this.fetchCategories().catch((error) => console.log('fetchCategories error ', error));
+    this.fetchQuestionSets().catch((error) => console.log('fetchQuestionSets error ', error));
+    this.fetchQuestions().catch((error) => console.log('fetchQuestions error', error));
     makeAutoObservable(this);
   }
 
-  setChosenMode = (chosenMode) => this.chosenMode = chosenMode;
+  setIsAnswerHasChosen = (bool) => {
+    this.isAnswerHasChosen = bool;
+  };
+
+  setChosenMode = (chosenMode) => {
+    this.chosenMode = chosenMode;
+  };
 
-  setCurrentQuestion = (question) => this.currentQuestion = question;
+  setCurrentQuestion = (question) => {
+    this.currentQuestion = question;
+  };
 
   setCurrentAnswers = () => {
     const [{answers}] = this.questions.filter((question) => question.text === this.currentQuestion.text);
-    const [{createdAt, id, updatedAt, ...answersArr}] = answers;
+    const [{
+      createdAt, id, updatedAt, ...answersArr
+    }] = answers;
     this.currentAnswers = Object.values(answersArr);
-  }
+  };
 
-  setChosenQuestionsSet = (chosenQuestionsSet) => this.chosenQuestionsSet = chosenQuestionsSet;
+  setChosenQuestionsSet = (chosenQuestionsSet) => {
+    this.chosenQuestionsSet = chosenQuestionsSet;
+  };
 
-  setCurrentCategory = (category) => this.currentCategory = category;
+  setCurrentCategory = (category) => {
+    this.currentCategory = category;
+  };
 
-  setShowHelpDialog = (bool) => this.showHelpDialog = bool;
+  setShowHelpDialog = (bool) => {
+    this.showHelpDialog = bool;
+  };
 
   getQuestions = (chosenQuestionsSetName) =>
     this.questions.filter(({questions_sets: [question]}) => question?.name === chosenQuestionsSetName);
 
-  getQuestionsSetsNames = (categoryTopic) =>
+  getFilteredQuestionsSets = (categoryTopic) =>
     this.questionSets.filter((questionsSet) => questionsSet.topic.name === categoryTopic.item.name);
```

Comment:
this is a bad practice I think (though you return filtered questions, it's bad that this inline function is actually written in 2 lines, so maybe wrap it with { return ... })